core/src/main/java/io/questdb/cairo/mv/DefinitionFileReader.java

```diff
@@ -0,0 +1,324 @@
+    }
+
+    @Override
+    public void close() throws IOException {
+        clear();
+        memory = null;
+        file = null;
+    }
+
+    public BlocksCursor getCursor() {
+        long regionLength;
+        while (true) {
+            long currentVersion = getVersionVolatile();
+            final long regionOffset = HEADER_SIZE + file.getLong(getRegionOffsetOffset(currentVersion));
+            regionLength = file.getLong(getRegionLengthOffset(currentVersion));
+
+            final long fileBaseAddress = file.getPageAddress(0);
+            final long memoryBaseAddress = memory.resize(regionLength);
+            Vect.memcpy(memoryBaseAddress, fileBaseAddress + regionOffset, regionLength);
+
+            if (currentVersion == getVersionVolatile()) {
+                break;
+            }
+        }
+
+        final long memoryBaseAddress = memory.getPageAddress(0);
+        final long checksumAddress = memoryBaseAddress + REGION_HEADER_SIZE + REGION_BLOCK_COUNT_OFFSET;
+        final long checksumSize = regionLength - REGION_HEADER_SIZE - REGION_BLOCK_COUNT_OFFSET;
+        final int checksum = getChecksum(checksumAddress, checksumSize);
+        final int expectedChecksum = memory.getInt(REGION_CHECKSUM_OFFSET);
+
+        if (checksum != expectedChecksum) {
+            throw CairoException.critical(0)
+                    .put("Checksum mismatch [expected=")
+                    .put(expectedChecksum)
+                    .put(", actual=")
+                    .put(checksum)
+                    .put(']');
+        }
+
+        final int blockCount = memory.getInt(REGION_BLOCK_COUNT_OFFSET);
+        assert blockCount > 0;
+        final long blocksLength = regionLength - REGION_HEADER_SIZE;
+        blocksCursor.of(blockCount, REGION_HEADER_SIZE, blocksLength);
+        return blocksCursor;
+    }
+
+    public long getVersionVolatile() {
+        return Unsafe.getUnsafe().getLongVolatile(
+                null,
+                file.getPageAddress(0) + HEADER_VERSION_OFFSET
+        );
+    }
+
+    public void of(@Transient final LPSZ path) {
+        clear();
+        final long pageSize = ff.getPageSize();
+        if (!ff.exists(path)) {
+            throw CairoException.critical(CairoException.ERRNO_FILE_DOES_NOT_EXIST)
+                    .put("Cannot find file: ").put(path);
+        }
+
+        if (ff.length(path) < HEADER_SIZE) {
+            throw CairoException.critical(0).put("Expected at least 1 block");
+        }
+
+        if (file == null) {
+            file = new MemoryCMRImpl();
+        }
+        file.of(ff, path, pageSize, ff.length(path), MemoryTag.MMAP_DEFAULT);
+        final long version = getVersionVolatile();
+        if (version == 0) {
+            throw CairoException.critical(0).put("File is empty");
+        }
+
+        if (memory == null) {
+            memory = new MemoryCARWImpl(pageSize, Integer.MAX_VALUE, MemoryTag.NATIVE_DEFAULT);
+        }
+    }
+
+    public class BlocksCursor {
```

Comment:
nit: should be `BlockCursor` similar to `RecordCursor`, not `RecordsCursor`

Reply:
renamed